### DISCUSSION TITLE
Scout search screen bug

### DISCUSSION THREAD

**Discussion by denpriler** [orchidsoftware/platform#2680] [Ideas]
Here is search screen bug on latest Laravel 10 package.
![image](https://github.com/orchidsoftware/platform/assets/113289986/8c9e4cbc-7964-40e7-9a28-56edd12195cc)
The problem is in SearchScreen class. Method 'get" of session() helper does not takes second argument for default values so $type variable becomes null instead of first available search model. It throws exception.
![image](https://github.com/orchidsoftware/platform/assets/113289986/1734bf76-1d3e-442f-bed3-2a1bebca4ed4)
So the fix was in copying SearchScreen to App\Orchid folder and redefining platform.search route.
![image](https://github.com/orchidsoftware/platform/assets/113289986/7ed81904-d725-4312-9787-d69f84eac86c)
Also after some debug I've found that the SearchLayout which generates radio buttons for choosing model filter also has bug.
Instance of Radio field class does not sets "checked" html attribute to needed radio input. I even tried with manually passing "true" boolean value to "checked" method despite is has "true" value as argument by default.
![image](https://github.com/orchidsoftware/platform/assets/113289986/ce863dad-12fb-426b-9ad7-4ef0918190d6)

This is not first bug I've got in process of creating very simple CRUD application. Please, fix this.

**Comment by ysalitrynskyi** [comment]
This bug still exists, for all radio buttons on the website. 
Temporary fix for search:

1) Method 1: If you have only 1 search model.

> Create: `resources/views/vendor/platform/partials/result.blade.php`
> Copy content from: `vendor/orchid/platform/resources/views/partials/result.blade.php`
> 
> Replace:
> `{!! $radios !!}`
> 
> With:
> `{!! str_contains($radios, 'checked') ? $radios : preg_replace('/'.preg_quote('<input', '/').'/', '<input checked', $radios, 1) !!}`

2) Method 2: Change to dropdown

> Copy files to:
> `app/Http/Layouts/SearchLayout.php`
> `app/Http/Screens/SearchScreen.php`
> 
> Make sure to change namespaces.
> 
> Add to /routes/platform.php:
> 
> ```
> use App\Http\Screens\SearchScreen;
> ....
> Route::screen('search/{query}', SearchScreen::class)
>     ->name('search')
>     ->breadcrumbs(fn (Trail $trail, string $query) => $trail->parent('platform.index')
>         ->push(__('Search'))
>         ->push($query));
> ```
> 
> In SearchLayout change replace:
> 
> ```
> $layouts = Dashboard::getSearch()
>             ->map(static function (Model $model) use ($searchModel) {
>                 $radio = Radio::make('type')
>                     ->value(get_class($model))
>                     ->horizontal()
>                     ->clear()
>                     ->placeholder($model->presenter()->label());
> 
>                 if ($model instanceof $searchModel) {
>                     $radio->checked();
>                 }
> 
>                 return $radio;
>             });
> 
>         $layouts->prepend(Label::make('search')->clear()->title(__('Choose record type:')));
> 
>         return [
>             Group::make($layouts->all())->autoWidth(),
>         ];
> ```
> 
> With:
> 
> ```
> $values = Dashboard::getSearch()
>             ->map(static function (Model $model) use ($searchModel) {
>                 return [
>                     get_class($model) => $model->presenter()->label(),
>                 ];
>             });
>         $layouts = Select::make('type')
>             ->title(__('Choose record type:'))
>             ->options(array_reduce($values->all(), 'array_merge', []))
>             ->value(get_class($searchModel));
> 
>         return [
>             Group::make([$layouts])->fullWidth(),
>         ];
> ```

Hope it will help somebody.
